pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -221,18 +267,78 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func createNameServerQuery(c client.Client, logger log.FieldLogger) nameserver.Query {
-	awsCredsSecretName := os.Getenv(constants.ExternalDNSAWSCredsEnvVar)
-	if awsCredsSecretName != "" {
-		logger.Infof("using aws creds for external DNS stored in %q secret", awsCredsSecretName)
-		return nameserver.NewAWSQuery(c, awsCredsSecretName)
+func createNameServerQuery(c client.Client, logger log.FieldLogger, managedDomain hivev1.ManageDNSConfig) nameserver.Query {
+	if managedDomain.AWS != nil {
+		secretName := managedDomain.AWS.CredentialsSecretRef.Name
+		logger.Infof("using aws creds for managed domains stored in %q secret", secretName)
+		return nameserver.NewAWSQuery(c, secretName)
+	}
+	if managedDomain.GCP != nil {
+		secretName := managedDomain.GCP.CredentialsSecretRef.Name
+		logger.Infof("using gcp creds for managed domain stored in %q secret", secretName)
+		return nameserver.NewGCPQuery(c, secretName)
+	}
+	logger.Error("unsupported cloud for managing DNS")
+	return nil
+}
+
+func updateParentLinkCreatedCondition(c client.Client, logger log.FieldLogger, dnsZone *hivev1.DNSZone, created bool, nameServers sets.String) (bool, error) {
+	var status corev1.ConditionStatus
+	var reason string
+	var message string
+	if created {
+		status = corev1.ConditionTrue
+		reason = "ParentLinkCreated"
+		message = fmt.Sprintf("Parent link created for name servers %s", nameServers)
+	} else {
+		status = corev1.ConditionFalse
+		reason = "ParentLinkNotCreated"
+		message = "Parent link has not been created"
 	}
 
-	gcpCredsSecretName := os.Getenv(constants.ExternalDNSGCPCredsEnvVar)
-	if gcpCredsSecretName != "" {
-		logger.Infof("using gcp creds for external DNS stored in %q secret", gcpCredsSecretName)
-		return nameserver.NewGCPQuery(c, gcpCredsSecretName)
+	return updateCondition(c, logger, dnsZone, hivev1.ParentLinkCreatedCondition, status, reason, message)
+}
+
+func updateMissingDomainClientCondition(c client.Client, logger log.FieldLogger, dnsZone *hivev1.DNSZone, missing bool) (bool, error) {
+	var status corev1.ConditionStatus
+	var reason string
+	var message string
+	if missing {
+		status = corev1.ConditionTrue
+		reason = "MissingDomainClient"
+		message = "Failed to find NameServer client for domain"
```

Comment:
The messages for this condition should be understandable to the end user. The message should talk about there not be a managed DNS domain rather than a NameServer client.